modules/exploits/multi/http/wp_plugin_sp_project_document_rce.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress Plugin SP Project and Document - Authenticated Remote Code Execution',
+        'Description' => %q{
+          This module allows an attacker with a privileged Wordpress account to launch a reverse shell
+          due to an arbitrary file upload vulnerability in Wordpress plugin SP Project & Document < 4.22.
+          This is due to an incorrect check of the uploaded file extension which should be of SGBP type.
+          Indeed, even if it is not possible to upload `.php` files, the security check only works with lowercase
+          making possible to upload `.pHP` files for instance.
+          Finally, the uploaded payload can be triggered by a call to `/wp-content/uploads/sp-client-document-manager/<user_id>/<random_payload_name>.php`
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'RON JOST', # Exploit-db
+            'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+          ],
+        'References' =>
+          [
+            %w[EDB 50115],
+            %w[CVE 2021-24347]
+          ],
+        'Platform' => ['php'],
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+        [
+          ['Wordpress SP Project & Document < 4.22', {}]
+        ],
+        'Privileged' => true,
```

Comment:
Should be set to false since you get privileges as the `www-data` user:

```suggestion
        'Privileged' => false,
```